connector/interface.go

```diff
@@ -60,6 +60,11 @@ type ConnectorConfig interface {
 	Connector(ns url.URL, loginFunc oidc.LoginFunc, tpls *template.Template) (Connector, error)
 }
 
+// Groups is an optional interface implemented by a Connector.
+type Groups interface {
+	Groups(fullUserID string) ([]string, error)
```

Comment:
I see. Yeah, lets keep them separate, and just try to find better naming. I'm not 100% on embedding the other interface either now, because I can imagine not every grouping implementation relies on a particular connector (the group implementation works for _all_ connectors, etc).